includes/sanitizers/class-amp-form-sanitizer.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Class AMP_Form_Sanitizer.
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_Form_Sanitizer
+ *
+ * Strips and corrects attributes in forms.
+ */
+class AMP_Form_Sanitizer extends AMP_Base_Sanitizer {
+
+	/**
+	 * Tag.
+	 *
+	 * @var string HTML <form> tag to identify and process.
+	 *
+	 * @since 0.2
+	 */
+	public static $tag = 'form';
+
+	/**
+	 * Sanitize the <img> elements from the HTML contained in this instance's DOMDocument.
+	 *
+	 * @since 0.2
+	 */
+	public function sanitize() {
+
+		/**
+		 * Node list.
+		 *
+		 * @var DOMNodeList $node
+		 */
+		$nodes     = $this->dom->getElementsByTagName( self::$tag );
+		$num_nodes = $nodes->length;
+
+		if ( 0 === $num_nodes ) {
+			return;
+		}
+
+		for ( $i = $num_nodes - 1; $i >= 0; $i-- ) {
+			$node = $nodes->item( $i );
+			if ( ! $node instanceof DOMElement ) {
+				continue;
+			}
+
+			if ( ! $node->hasAttribute( 'action' ) || '' === $node->getAttribute( 'action' ) ) {
+				$node->parentNode->removeChild( $node );
```

Comment:
I don't think this is right. If `method` is `get` and there is no `action`, then what this really should do I think is just set the `action` to be `esc_url_raw( 'https://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'] )`, because the default `action` for a `form` is the URL that the form appears on.